website/content/projects/2019/spec.md

```diff
@@ -0,0 +1,386 @@
+
+
+
+*   “We ultimately decided we don’t need a circle module after all. I’m closing this issue, but will reopen if we decide to revisit this”
+*   “The circle module has been implemented, except for the intersection function, which turned out to be more complex than anticipated. I am closing this issue, and have created a separate task (#5) for implementing circle intersections”
+
+
+### Pull Requests to master
+
+Like in many software projects, the master branch of our Git repository should only contain stable code that compiles, runs, and passes all tests. This is enforced by making the master branch a _protected_ branch, meaning that only the instructors and TAs have the ability to directly push to that branch. Everyone else must make a pull request to get code merged into the master branch.
+
+Because the master branch should contain stable code, pull requests to master must be carefully reviewed before they are approved. If you want to merge code into master, your code must be reviewed by (1) someone in your team, (2) someone in another team, and (3) your senior developer. When your code substantially affects or uses code from another component, then the reviewers from another team must specifically be from the team whose code is being used or affected.
+
+A pull request should contain a relatively self-contained piece of work.  In fact, it is not uncommon for a task (or, at most, 2-3 tasks) to result in a pull request. We encourage you to touch base with your senior developer before creating your first pull requests, to determine whether the work you want to merge merits a pull request (or, similarly, whether it needs to be broken up into multiple pull requests).
+
+Like a task, a pull request should have a descriptive title, a descriptive summary, updates on the progress of the pull request (especially since changes may need to be done based on the code reviews), and a closing statement. The assignees for the pull request must be everyone who contributed to the code that is being merged. If a pull request is approved, all assignees and reviewers will receive the points for that pull request.
+
+Take into account that the final outcome of a pull request could be for that pull request to be denied. If this happens, it will not impact your score (but you will also not receive points for that pull request)
+
+
+### Code Reviews
+
+You are welcome to create pull requests that do not affect the master branch, and to use these as a mechanism to do code reviews. For example, suppose the Game State team has a “game-state” branch where they do most of their work, and you are working on a “room” branch, writing a module to model rooms. Once you’re done, you could do a pull request from “room” to “game-state” instead of directly merging the branch. This would allow you to request a code review from someone else on your team (and to have a record of that code review in the pull request).
```

Comment:
`s/could do a pull request/should do a pull request/`

At least, IMO

Reply:
SGTM :shipit: